### ISSUE TITLE
azurerm_cosmosdb_sql_container_resource is failing to update on migrated containers with system assigned partition key

### ISSUE THREAD

**Issue by thereallukl** [hashicorp/terraform-provider-azurerm#18204]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.7

### AzureRM Provider Version

3.9.0

### Affected Resource(s)/Data Source(s)

azurerm_cosmosdb_sql_container_resource

### Terraform Configuration Files

```hcl
resource "azurerm_cosmosdb_sql_container" "main" {
  account_name           = var.account_name
  resource_group_name    = var.resource_group_name
  database_name          = var.database_name
  name                   = var.name
  partition_key_path     = var.partition_key_path
  partition_key_version  = var.partition_key_version
}
```

### Debug Output/Panic Output

```shell
Error: waiting on create/update future for Cosmos SQL Container "<<edited_container_name>>" (Account: "<<edited_account_name>>", Database: "<<edited_db_name>>"): Code="BadRequest" Message="Message: {\"code\":\"BadRequest\",\"message\":\"Message: {\\\"Errors\\\":[\\\"This is a migrated fixed collection and document collection partition key cannot be changed, Learn more: https:\\\\/\\\\/aka.ms\\\\/migrated-non-partitioned-collection\\\"]}
```

### Expected Behaviour

Provider should read value of `properties.resource.partitionKey.systemKey`, save it internally in state and reuse for each update operation.

### Actual Behaviour

`properties.resource.partitionKey.systemKey` is skipped with update requests, which for migrated containers always ends up BadRequest error as `properties.resource.partitionKey` cannot be changed after creation.

### Steps to Reproduce

1. Import existing container with migrated non partitioned collection to TF state
2. terraform apply

### References

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2022-05-15/cosmos-db.json#L7472

**Comment by tomabg** [comment]
same error when creating resource "azurerm_cosmosdb_sql_database"

2022-09-02T08:26:10.9449942Z ##[error]Terraform command 'apply' failed with exit code '1'.
2022-09-02T08:26:10.9557349Z ##[error][31m╷[0m[0m
mwaiting on create/update future for Sql Database: (Name "XXX" / Database Account Name "XXXX" / Resource Group "XXX"): Code="ServiceUnavailable" Message="Service is currently unavailable. More info: https://aka.ms/cosmosdb-tsg-service-unavailable\r\nActivityId: XXX, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0"[0m
[31m│[0m [0m

**Comment by katbyte** [comment]
Thank you for taking the time to report this issue, and apologies that it has sat without a resolution for so long.

This bug was reported against v3.x of the AzureRM provider. The provider has since moved on to v5.x, and the affected functionality has often been significantly reworked across those major releases, so bug reports from legacy major versions prior to v4.0 are generally no longer actionable as written.

As part of a cleanup of historical issues we are closing bug reports that target legacy provider versions. If you are still running into this behaviour on the current major version, please open a new issue with up-to-date reproduction details (feel free to reference this one) so it can be triaged with current context.

Thank you again for the report, and for your patience.
